cpp/src/arrow/util/compression.cc

```diff
@@ -48,13 +48,42 @@
 namespace arrow {
 namespace util {
 
+int32_t GetCompressionCodecDefaultCompressionLevel(Compression::type t) {
```

Comment:
I think it doesn't because then this would add a dependency on the implementation of the codec. The implementation would only be visible if the library is compiled in and this would introduce a lot of ifdefs (ifdef ARROW_WITH_ZLIB, etc).

However, I can change it as you requested. I'm probably not getting this patch in if I would be stubborn for everything.

Reply:
I decided to follow what @pitrou recommended by using numeric_limits<int>::min as a hint to let the implementation select whatever compression level it deems appropriate.

Reply:
I'm making this a static method... I will push changes to your branch